packages/esm-form-render-app/src/render/forms-render-test.component.tsx

```diff
@@ -74,6 +75,65 @@ function FormRenderTest() {
     setIsSchemaLoaded(false);
   };
 
+  const formValidator = () => {
+    if (defaultJson) {
+      const parsedForm = typeof defaultJson == 'string' ? JSON.parse(defaultJson) : defaultJson;
+
+      for (let i = 0; i < parsedForm.pages.length; i++) {
+        for (let j = 0; j < parsedForm.pages[i].sections.length; j++) {
+          for (let k = 0; k < parsedForm.pages[i].sections[j].questions.length; k++) {
+            
+            const questionObject = parsedForm.pages[i].sections[j].questions[k];
+            handleFormValidation(questionObject);
+          }
+        }
+      }
+    } else {
+      console.log('Empty form!');
+    }
+  };
+
+  const handleFormValidation = (conceptObject) => {
+    conceptObject.questionOptions.concept
+      ? openmrsFetch(`/ws/rest/v1/concept/${conceptObject.questionOptions.concept}`)
+          .then((response) => {
+            console.log(response.data);
+            dataTypeChecker(conceptObject, response);
+          })
+          .catch((error) => {
+            error.message.includes('404') &&
+              console.log(`❌ Concept UUID ${conceptObject.questionOptions.concept} not found`);
+          })
+      : console.log('❌ Question with no concept UUID: ', conceptObject.id);
+  };
+
+  const dataTypeChecker = (conceptObject, responseObject) => {
+    const dataTypes = ['Numeric', 'Coded', 'Text', 'Date', 'Datetime', 'Boolean', 'Rule'];
+
+    const renderTypes = [
+      ['number'],
+      ['select', 'checkbox', 'radio', 'toggle', 'content-switcher'],
+      ['text', 'textarea'],
+      ['date'],
+      ['datetime'],
+      ['toggle', 'select', 'radio', 'content-switcher'],
+      ['repeating', 'group']
+    ];
+
+    if (conceptObject.questionOptions.concept === responseObject.data.uuid) {
+
+      dataTypes.forEach((dataType, index) => {
+        responseObject.data.datatype.display === `${dataType}` &&
+        !renderTypes[index].includes(conceptObject.questionOptions.rendering) &&
+          console.log('❌ datatype mismatch');
+      });
+
+      console.log(
```

Comment:
I don't understand why this is necessary

Reply:
Not really necessary, just added to give more info for clear comparison at a glance.

Reply:
Has been removed.